### PR TITLE
download video gradually when block given

### PR DESCRIPTION
**Pull request by tdtds**

### CODE REVIEW

File: lib/niconico/video.rb [hunk 1]
```diff
@@ -86,7 +86,18 @@ def available?
 
     def get_video
       raise VideoUnavailableError unless available?
-      @agent.get_file(video_url)
+      unless block_given?
+          @agent.get_file(video_url)
+      else
+        cookies = @agent.cookie_jar.cookies('http://nicovideo.jp').map{|c|c.to_s}.join(';')
```
Comment (sorah) [inline lib/niconico/video.rb]: - `@agent.cookie_jar.cookies('http://nicovideo.jp')` can be replaced with `video_cookies` (defined below)
- `map(&:to_s)`

**Comment by sorah** [comment]
Otherwise LGTM. Please squash additinal changes to keep commit log clean.

**Comment by tdtds** [comment]
fixed. merge plz.

**Comment by sorah** [comment]
:sparkles:

**Pull request merged**